File: javascript/001.js
/**
 * Project Euler - Problem Solution 001
 * Copyright (c) Justin McGettigan. All rights reserved.
 * https://github.com/jwmcgettigan/project-euler-solutions
 */

const multipleOf = (num, multiple) => num % multiple == 0;
const sumofMultiples = (limit) => {
  let sum = 0;
  for(let x = 0; x < limit; x++) {
    if(multipleOf(x, 3) || multipleOf(x, 5)) {
      sum += x;
    }
  }
  return sum;
}
console.log(sumofMultiples(1000));

File: solutions/001/001.py
"""
Project Euler - Problem Solution 001
Copyright (c) Justin McGettigan. All rights reserved.
https://github.com/jwmcgettigan/project-euler-solutions
"""

def multipleOf(num, multiple):
  return num % multiple == 0

def sumOfMultiples(limit):
  return sum(x for x in range(limit) if multipleOf(x, 3) or multipleOf(x, 5))

if __name__ == "__main__":
	print(sumOfMultiples(1000))